### PR TITLE
[ASVideoNode] Maintain playback position between visibility changes

### PR DESCRIPTION
**Pull request by gazreese**
My current project uses ASVideoNode / ASVideoPlayerNode within a collection view. Unfortunately, when the videos scroll out of view and back in again the playback position is reset to the beginning. I feel that this is contrary to what users would expect.

This change maintains the current playback time in an ivar so that we can reestablish the ASVideoNode display at the same position as it was when it left view.

### CODE REVIEW

**Comment by gazreese** [comment]
Merged the master branch into mine again to resolve merge conflicts

File: AsyncDisplayKit/ASVideoNode.mm [hunk 1]
```diff
@@ -454,6 +460,9 @@ - (void)didExitVisibleState
   ASDN::MutexLocker l(__instanceLock__);
   
   if (_shouldBePlaying) {
+    if(_player != nil && CMTIME_IS_VALID(_player.currentTime)) {
+      _lastPlaybackTime = _player.currentTime;
+    }
     [self pause];
     _shouldBePlaying = YES;
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/ASVideoNode.mm]: Small things: Space after `if` and let's pause before reading the current time.

**Comment by Adlai-Holler** [comment]
@gazreese Thanks for this improvement, and sorry for the long turnaround time on a review. One comment and we'll take it to the hoop!

**Comment by Adlai-Holler** [comment]
Excellent! Thank you @gazreese for making this happen.

**Pull request merged**